lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -1180,12 +1210,22 @@ private function isLimitQuery()
      * Converts this instance into an INSERT string in SQL.
      *
      * @return string
+     * @throws QueryException
      */
     private function getSQLForInsert()
     {
-        return 'INSERT INTO ' . $this->sqlParts['from']['table'] .
-        ' (' . implode(', ', array_keys($this->sqlParts['values'])) . ')' .
-        ' VALUES(' . implode(', ', $this->sqlParts['values']) . ')';
+        $statement = 'INSERT INTO ' . $this->sqlParts['into']['table'];
+        if( !empty($this->sqlParts['values']) && count( array_keys($this->sqlParts['values']) ) > 0 ) {
+            $statement .= ' (' . implode(', ', array_keys($this->sqlParts['values'])) . ')';
+        }
+
+        if( !empty($this->sqlParts['select'] ) ) {
+            $statement .= ' (' . $this->getSQLForSelect() . ')';
```

Comment:
The parentheses around `SELECT` are not needed. At least in most cases.

Reply:
I know, but if it's a slightly more complex statement then I'm on the safe side

Reply:
> slightly more complex statement

If you have such a scenario, please add a test for it